### PR TITLE
Metrics : Added IP family used for the cluster as prometheus metric.

### PR DESCRIPTION
**Pull request by gvbalaji**
Signed-off-by: Balaji Varadaraju <bvaradar@redhat.com>
Added IP family used for the cluster as prometheus metric.

### CODE REVIEW

File: go-controller/pkg/metrics/ovn.go [hunk 1]
```diff
@@ -380,4 +382,33 @@ func RegisterOvnControllerMetrics() {
...
 	go coverageShowMetricsUpdater(ovnController)
+	// Set IP family metric
+	ipFamilyMetricsPublisher()
+}
+
+func ipFamilyMetricsPublisher() {
+	var metricIPFamily = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: MetricOvnkubeNamespace,
+		Subsystem: MetricOvnkubeSubsystemMaster,
+		Name:      "stack",
```
Comment (dcbw) [inline go-controller/pkg/metrics/ovn.go:393]: @gvbalaji maybe "ip_family" or "ip_families" not stack?

Comment (gvbalaji) [reply go-controller/pkg/metrics/ovn.go]: Thanks Dan. I had it as ip_family to start with but it was changed with some unintended copy-paste.:) I reverted to ip_family.

File: go-controller/pkg/metrics/ovn.go [hunk 2]
```diff
@@ -380,4 +382,33 @@ func RegisterOvnControllerMetrics() {
...
+	},
+		[]string{
+			"stack",
+		},
+	)
+	ovnRegistry.MustRegister(metricIPFamily)
+	metricIPFamily.WithLabelValues(getIPFamily()).Set(1)
+}
+
+func getIPFamily() string {
```
Comment (squeed) [inline go-controller/pkg/metrics/ovn.go:406]: Why not publish two series - IPv4 and IPv6, and set them both to 1 in the dual-stack case?

Comment (gvbalaji) [reply go-controller/pkg/metrics/ovn.go:406]: I can do it if that is a better way. Only issue is to know the IP family of the cluster
1. Users need to query two metrics
2. Then have the logic to determine if it is dual stack.

To me having a single metric is simpler to use. Please let me know.

Comment (squeed) [reply go-controller/pkg/metrics/ovn.go:406]: It depends on what the users might wish to query. Do they want to know "does this cluster support v6?" or "is this cluster dual-stack"?

I can see possible use-cases for both. For example, I might want to write a query "if the cluster is v6, alert if a node doesn't have an ipv6 address". That doesn't matter if the cluster is dual-stack.

Comment (gvbalaji) [reply go-controller/pkg/metrics/ovn.go:406]: That use is satisfied with the single metric as well. Isn't it? They will know  if the node needs to have one or both the IP addresses.

Comment (gvbalaji) [reply go-controller/pkg/metrics/ovn.go:406]: Can we please get a conclusion on this PR?

**Pull request closed without merging**